sdk/test-utils/recorder-new/src/matcher.ts

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { createPipelineRequest, HttpClient } from "@azure/core-rest-pipeline";
+import { paths } from "./utils/paths";
+import { RecorderError } from "./utils/utils";
+
+export type Matcher = "HeaderlessMatcher" | "BodilessMatcher";
+
+export async function setMatcher(
```

Comment:
I find it a bit confusing that this function and the outer one are named the same 🤔 are there alternatives?

Reply:
Hmm, I see where you're coming from, and considered it initially. In the end I couldn't come up with a different name for one of the two that I was happy with. Open to suggestions, but IMHO the fact that one is a method on a class and the other is a standalone function provides enough distinction.

Reply:
Specifically, because this is an internal method and not exposed to the public.